### PR TITLE
[deprecated]: [DO NOT MERGE] Adding test case for installing openebs control plane components on specific nodes

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <uditgaurav@gmail.com>

**_Issue it resolve:_** https://github.com/mayadata-io/oep/issues/219

**_This PR is for:_**

 - Adding test case for installing openebs control plane components on specific nodes

**_This issue is for:_**
- Adding test case for installing openebs control plane components on specific nodes

**_Prerequisites:_**

- The cluster should be a minimum of **three** nodes.
- It should have enough `CPU` and `Memory` to install openebs. We can use at least the `n1-standard-1` size of the cluster.

**_Details:_**
 - This issue is for creating the test cases for openebs control plane installation on the specific nodes. It can be done by labeling the nodes before installing the openebs on it. So basically we will test that the labeling feature is working file with respective of control plane components. Control plane components include all the openebs components expect NDM daemonset.

- The tasks involved for creating the test case will be - we will labels nodes as shown in the table.

<table style="width:100%">
  <tr>
    <th>OpenEBS components</th>
    <th>Node-1</th>
    <th>Node-2</th>
    <th>Node-3</th>
  </tr>
  <tr>
    <td><b>Control plane components</b></td>
    <td>True</td>
    <td>Flase</td>
    <td>False</td>
  </tr>
  <tr>
    <td><b>Data plane components</b></td>
    <td>False</td>
    <td>True</td>
    <td>True</td>
  </tr>
</table>

- Now we will test if the `node-1` id labeled with control plane component or not also `node-2` and `node-3` should not be labeled with the control plane components.

**_Logs of the job:_**
- [X] [logs1.txt](https://github.com/mayadata-io/oep/files/4227789/logs1.txt)

### CODE REVIEW

File: litmus/director/openebs-control-plane-test/run_litmus_test.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  template:
+    metadata:
+      name: litmus
+      labels:
+        app: openebs-control-plane-test-litmus
+    spec:
+      serviceAccountName: litmus
+      restartPolicy: Never
+      volumes:
+      - name: secret-volume
```
Comment (AmitKumarDas) [inline litmus/director/openebs-control-plane-test/run_litmus_test.yml]: need to explain about secret in the readme

File: litmus/director/openebs-control-plane-test/run_litmus_test.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        secret:
+          secretName: director-user-pass
+      containers:
+      - name: ansibletest
+        image: mayadataio/dop-validator:ci
```
Comment (AmitKumarDas) [inline litmus/director/openebs-control-plane-test/run_litmus_test.yml]: why are we using ci as tag? we should ideally make use of release versions

Comment (AmitKumarDas) [reply litmus/director/openebs-control-plane-test/run_litmus_test.yml]: README should explain the role of dop-validator

File: litmus/director/openebs-control-plane-test/run_litmus_test.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            value: ''
+
+          ## Enter Memory resource limit
+          - name: MEMORY_RESOURCE_LIMIT
+            value: ''
+          
+          ## It will have values basic/advance
+          ## Note: For basic installation only template value slould be provided
+          ## For adding any additional value - change the installation mode to advance
+          - name: INSTALLATION_MODE
```
Comment (AmitKumarDas) [inline litmus/director/openebs-control-plane-test/run_litmus_test.yml]: Should we always test basic mode ? Why not other modes? cc @ajeshbaby

File: litmus/director/openebs-control-plane-test/run_litmus_test.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            value: 'basic'
+
+          - name: ANSIBLE_STDOUT_CALLBACK
+            value: 'default'  
+
+        command: ["/bin/bash"]
+        args: ["-c", "ansible-playbook ./litmus/director/openebs-control-plane-test/test.yml -i /etc/ansible/hosts -v; exit 0"]
```
Comment (AmitKumarDas) [inline litmus/director/openebs-control-plane-test/run_litmus_test.yml]: can we rearrange the args in a list format?

File: litmus/director/openebs-control-plane-test/test.yml [hunk 5]
```diff
@@ -0,0 +1,264 @@
+---
+- hosts: localhost
```
Comment (AmitKumarDas) [inline litmus/director/openebs-control-plane-test/test.yml]: Need to explain in code comments the role of hosts & connections.
Need to explain if the code is run from within a container or outside a container & so on.

**Comment by uditgaurav** [comment]
Closing this PR due to some Github issue in the PR

**Pull request closed without merging**